src/jsonprovider/jsonprovider.cpp

```diff
@@ -0,0 +1,257 @@
+/*
+ * Copyright (C) 2024 EPAM Systems, Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <utils/exception.hpp>
+#include <utils/json.hpp>
+
+#include <Poco/JSON/Parser.h>
+
+#include "jsonprovider/jsonprovider.hpp"
+
+namespace aos::common::jsonprovider {
+
+static DeviceInfo ParseDeviceInfo(const utils::CaseInsensitiveObjectWrapper& object)
```

Comment:
Put statics in unnamed namespace?

Could you use same pattern for naming as for jsonprovider interface. I mean:

DeviceInfoToJSON
DeviceInfoFromJSON

etc.